### PR TITLE
V0.6 pgp handler

### PR DESCRIPTION
**Pull request by csoler**
Turns PGPHandler into an abstract class that is derived as OpenPGPSDKHandler to provide a specific implementation for openpgpsdk. This should make it easier to switch to any other PGP package. Also got rid of getAuthGPG(), using static members for AuthGPG instead.

### CODE REVIEW

File: libretroshare/src/gossipdiscovery/p3gossipdiscovery.cc [hunk 1]
```diff
@@ -219,7 +219,7 @@ void p3discovery2::removeFriend(const RsPeerId &sslId)
 		std::cerr << std::endl;
 #endif
 		/* pgp peer without any ssl entries -> check if they are still a real friend */
-		if (!(AuthGPG::getAuthGPG()->isGPGAccepted(pgpId)))
+        if (!(AuthGPG::isGPGAccepted(pgpId)))
```
Comment (G10h4ck) [inline libretroshare/src/gossipdiscovery/p3gossipdiscovery.cc]: Now that you are refactoring this please rename `isGPGAccepted` into `isPgpAccepted` note the letters changed not just the casing ;)

File: libretroshare/src/pgp/pgphandler.cc [hunk 2]
```diff
@@ -33,1516 +33,82 @@
...
-#else
-	ops_memory_t *memres = ops_sign_buf(data,len,OPS_SIG_BINARY,OPS_HASH_SHA1,secret_key,ops_false,ops_false,not_raw,not_raw) ;
-#endif
+	for(std::map<RsPgpId,PGPCertificateInfo>::const_iterator it(_public_keyring_map.begin()); it != _public_keyring_map.end(); ++it)
+	{
+		std::cerr << "PGP Key: " << it->first.toStdString() << std::endl;
 
-	if(!memres)
-		return false ;
+		std::cerr << "\tName          : " <<  it->second._name << std::endl;
```
Comment (G10h4ck) [inline libretroshare/src/pgp/pgphandler.cc]: Isn't better to use `RS_DBG() <<` and friends here so it works also on android without black magic?

**Review by G10h4ck — CHANGES_REQUESTED** [review]

**Comment by G10h4ck** [comment]
Beside the inline comments, shouldn't we use a global pointer + methods like we do for most services like `rsFiles->someMethod()` this way we can easily expose the functionality also to JSON API, moreover i think it is easier to integrate other PGP implementations that way instead of static methods, as another implementation may need to keep some runtime state which is more difficult to do within static stuff.

**Comment by csoler** [comment]
I actually hesitated between the two options. This one has some advantages as well, but otherwise the pointer method would be more consistent with the rest of the code.

**Comment by G10h4ck** [comment]
I agree on keeping it homogeneous with the rest of the code, so please switch to global pointer + methods style now that your are still in time ;)
When I started coding into RS having so different patterns and anti-patterns into different parts of code added much to the entry barrier.

File: libretroshare/src/gossipdiscovery/p3gossipdiscovery.cc (hunk 1 above)
Comment (csoler) [reply libretroshare/src/gossipdiscovery/p3gossipdiscovery.cc]: done

**Comment by csoler** [comment]
Well actually looking at the code AuthPGP is not a structure that we may expose at the API level. So there's no problem making it static.

File: libretroshare/src/pgp/pgphandler.cc (hunk 2 above)
Comment (G10h4ck) [reply libretroshare/src/pgp/pgphandler.cc]: ping

**Comment by csoler** [comment]
ok I fixed both conflicts and std::cerr problems.

**Pull request merged**